src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,268 @@
+  const nodeDepth = context?.nodeDepth ?? 0;
+  const subNodeIds = context?.subNodeIds ?? [];
+  const parentNodeId = context?.parentNodeId;
+  const groupSize = context?.groupSize ?? 0;
+  const indexInGroup = context?.indexInGroup ?? 0;
+
+  const [isFocused, setIsFocused] = React.useState(false);
+  const nodeRef = React.useRef<HTMLLIElement>(null);
+
+  const styleDepth = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        if (isExpanded) {
+          onExpanded(nodeId, false);
+        } else if (parentNodeId) {
+          const parentNode = nodeRef.current?.ownerDocument.querySelector(
+            `#${parentNodeId}`,
+          ) as HTMLElement;
+          parentNode?.focus();
+        }
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        if (!isExpanded && hasSubNodes) {
+          onExpanded(nodeId, true);
+        } else if (subNodeIds.length > 0) {
+          const subNodes = Array.from(
+            nodeRef.current?.ownerDocument.querySelectorAll(
+              `${subNodeIds.map((id) => `#${id}`).join(',')}`,
+            ) || [],
+          ).filter(
+            (el) => el.getAttribute('aria-disabled') !== 'true',
+          ) as HTMLElement[];
+          subNodes[0]?.focus();
+        }
+        break;
+      case ' ':
+      case 'Spacebar':
+      case 'Enter':
+        // Ignore if it is called on the element inside, e.g. checkbox or expander
+        if (event.target !== nodeRef.current) {
+          break;
+        }
+        event.preventDefault();
+        if (!isDisabled) {
+          onSelected?.(nodeId, !isSelected);
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <li
+      role='treeitem'
+      id={nodeId}
+      aria-expanded={hasSubNodes ? isExpanded : undefined}
+      aria-disabled={isDisabled}
+      aria-selected={isSelected}
+      aria-level={nodeDepth + 1}
+      aria-setsize={groupSize}
+      aria-posinset={indexInGroup + 1}
+      tabIndex={-1}
+      onFocus={(e) => {
+        setIsFocused(true);
+        // Prevents from triggering onFocus on parent Tree
+        e.stopPropagation();
+      }}
+      onBlur={() => {
+        setIsFocused(false);
+      }}
+      ref={nodeRef}
+      onKeyDown={onKeyDown}
+      {...rest}
+    >
+      {
+        <div
+          className={cx(
+            'iui-tree-node',
+            {
+              'iui-active': isSelected,
+              'iui-disabled': isDisabled,
+            },
+            className,
+          )}
+          style={styleDepth}
+          onClick={() => !isDisabled && onSelected?.(nodeId, !isSelected)}
+          onDoubleClick={() =>
```

Comment:
I am thinking to remove double click, because it firstly selects the node, then deselects it:
![tree-node-dbclick](https://user-images.githubusercontent.com/36186912/153851167-ad982f6e-2155-4fa0-9480-071f330e0f4f.gif)

And to fix this, we would need some hacky solutions to check whether it is one click or double click.

Reply:
I brought it up initially, but I also tohught we have that in table, which we dont. So I am ok to remove it.

Reply:
Removed double click handling